### PR TITLE
Merge Promise Guard with ESNEXT

### PR DESCRIPTION
**Pull request by Achie72**
JerryScript-DCO-1.0-Signed-off-by: Bela Toth tbela@inf.u-szeged.hu

### CODE REVIEW

**Review by rerobika — APPROVED** [review]

File: jerry-core/config.h [hunk 1]
```diff
@@ -552,12 +548,8 @@
 || ((JERRY_BUILTIN_WEAKREF != 0) && (JERRY_BUILTIN_WEAKREF != 1))
 # error "Invalid value for JERRY_BUILTIN_WEAKREF macro."
 #endif
-#if !defined (JERRY_BUILTIN_PROMISE) \
-|| ((JERRY_BUILTIN_PROMISE != 0) && (JERRY_BUILTIN_PROMISE != 1))
-# error "Invalid value for JERRY_BUILTIN_PROMISE macro."
-#endif
 #if !defined (JERRY_BUILTIN_PROXY) \
-|| ((JERRY_BUILTIN_PROXY != 0) && (JERRY_BUILTIN_PROXY != 1))
+|| ((JERRY_ESNEXT != 0) && (JERRY_BUILTIN_PROXY != 1))
```
Comment (galpeter) [inline jerry-core/config.h:552]: Is this really correct? Do we want to change the proxy also?

Comment (Achie72) [reply jerry-core/config.h:552]: No, that's something i swapped by mistake, which should also explain the Proxy errors that the build option tests are throwing.

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -2681,7 +2681,7 @@ jerry_value_is_object (const jerry_value_t value)
 Returns whether the given `jerry_value_t` is a promise value.
 
 *Notes*:
-- This API depends on a build option (`JERRY_BUILTIN_PROMISE`) and can be checked
+- This API depends on a build option (`JERRY_ESNEXT`) and can be checked
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:2684]: This dependency change must be documented in this api ref. Please add a "Changed in version [[NEXT_RELEASE]]" entry (after the "new in version" entry).

File: jerry-core/ecma/operations/ecma-objects.c [hunk 3]
```diff
@@ -2849,10 +2847,10 @@ static const uint16_t ecma_class_object_magic_string_id[] =
 #if JERRY_MODULE_SYSTEM
   LIT_MAGIC_STRING_MODULE_UL, /**< magic string id of ECMA_OBJECT_CLASS_MODULE */
 #endif
-#if JERRY_BUILTIN_PROMISE
+#if JERRY_ESNEXT
   LIT_MAGIC_STRING_PROMISE_UL, /**< magic string id of ECMA_OBJECT_CLASS_PROMISE */
   LIT_MAGIC_STRING_OBJECT_UL, /**< magic string id of ECMA_OBJECT_CLASS_PROMISE_CAPABILITY */
-#endif /* JERRY_BUILTIN_PROMISE */
+#endif
```
Comment (galpeter) [inline jerry-core/ecma/operations/ecma-objects.c:2853]: Missing #endif guard comment.

**Review by galpeter — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM after minor changes

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**